lib/semantics/check-io.cc

```diff
@@ -15,12 +15,62 @@
 #include "check-io.h"
 #include "expression.h"
 #include "tools.h"
+#include "../common/format.h"
 #include "../parser/tools.h"
 
 namespace Fortran::semantics {
 
 // TODO: C1234, C1235 -- defined I/O constraints
 
+void IoChecker::Enter(
+    const parser::Statement<common::Indirection<parser::FormatStmt>> &stmt) {
+  if (!stmt.label.has_value()) {
+    context_.Say("Format statement must be labeled"_err_en_US);  // C1301
+  }
+  parser::CharBlock stmtCharBlock{stmt.source};
+  auto parenIndex{stmtCharBlock.ToString().find('(')};
+  parser::CharBlock formatCharBlock{parser::CharBlock(
+      stmtCharBlock.begin() + parenIndex, stmtCharBlock.size() - parenIndex)};
```

Comment:
No need for the second `parser::CharBlock` in this statement.

Reply:
Done; thanks.